src/domain/kaivuilmoitus/Attachments.tsx

```diff
@@ -0,0 +1,168 @@
+import FileUpload from '../../common/components/fileUpload/FileUpload';
+import styles from './BasicInfo.module.scss';
+import { Link, Notification } from 'hds-react';
+import TextArea from '../../common/components/textArea/TextArea';
+
+type Props = {
+  existingAttachments: ApplicationAttachmentMetadata[] | undefined;
+  attachmentsLoadError: boolean;
+  onFileUpload: (isUploading: boolean) => void;
+};
+
+function Attachments({ existingAttachments, attachmentsLoadError, onFileUpload }: Readonly<Props>) {
+  const queryClient = useQueryClient();
+  const { t } = useTranslation();
+  const { getValues } = useFormContext<KaivuilmoitusFormValues>();
+  const alluStatus = getValues('alluStatus');
+
+  function handleFileUpload(uploading: boolean) {
+    onFileUpload(uploading);
+    if (!uploading) {
+      queryClient.invalidateQueries('attachments');
+    }
+  }
+
+  const showDeleteButton = useCallback(() => {
+    return alluStatus === null;
+  }, [alluStatus]);
+
+  return (
+    <div>
+      <div className={styles.formInstructions}>
+        <Trans
+          i18nKey="kaivuilmoitusForm:liitteetJaLisatiedot:instructions"
+          components={{
+            a: (
+              <Link
+                href="https://www.hel.fi/static/hkr/luvat/tyyppikuvat/Tyyppikuvat.pdf"
+                openInNewTab
+              >
+                Helsingin kaupungin tyyppikuvat
+              </Link>
+            ),
+          }}
+        ></Trans>
+      </div>
+
+      <Box as="h2" className="heading-m" marginBottom="var(--spacing-s)">
+        {t('kaivuilmoitusForm:liitteetJaLisatiedot:trafficArrangementPlan')}
+      </Box>
+      <FileUpload
+        id="excavation-notification-file-upload-traffic-arrangement-plan"
+        accept=".pdf"
+        maxSize={104857600}
+        dragAndDrop
+        multiple
+        existingAttachments={existingAttachments?.filter(
+          (metadata) => metadata.attachmentType === 'LIIKENNEJARJESTELY',
+        )}
+        existingAttachmentsLoadError={attachmentsLoadError}
+        maxFilesNumber={20}
+        uploadFunction={({ file, abortSignal }) =>
+          uploadAttachment({
+            applicationId: getValues('id')!,
+            attachmentType: 'LIIKENNEJARJESTELY',
+            file,
+            abortSignal,
+          })
+        }
+        onUpload={handleFileUpload}
+        fileDownLoadFunction={(file) => getAttachmentFile(getValues('id')!, file.id)}
+        fileDeleteFunction={(file) =>
+          deleteAttachment({ applicationId: getValues('id'), attachmentId: file?.id })
+        }
+        onFileDelete={() => queryClient.invalidateQueries('attachments')}
+        showDeleteButtonForFile={showDeleteButton}
+      />
+
+      <Box as="h2" className="heading-m" marginBottom="var(--spacing-s)">
+        {t('kaivuilmoitusForm:liitteetJaLisatiedot:mandate')}
+      </Box>
+      <Notification
+        size="small"
+        type="info"
+        label="Mandate info"
```

Comment:
This label would be good to be in translation file even though it's not displayed because it's accessible to assistive technology.

Reply:
Good to know, thanks! I am not sure what sort of text should be in the label so I created a new text for it: "Tarkista valtakirja".

Reply:
That seems like a good text.